### PR TITLE
TagAction and multiple RoutingTags for routing logic

### PR DESCRIPTION
**Pull request by workgena**
Checklist:

- [x] create table "class4.tag_actions"
- [x] add references "tag_action_id" to: `customers_auth`, `numberlists`, `numberlist_items`, `routing_tag_detection_rules`
- [x] add `belongs_to :tag_action_id`  to affected models
- [x] ActiveAdmin: add tag_action-selectbox to affected to models
- [x] Prohibit removing "Routing Tag" used in any related resources(CustomersAuth, Nubmerlist, NubmerlistItem, RoutingTagDetectionRule, Destination, Dialpeer)
- [x] Migrate old data
- [x] API Admin: add readonly endpoint `tag_actions`
- [x] API Admin: add `has_one :tag_action` to affected models
- [x] Update API documentations
- [x] ActiveAdmin: verify "Copy" action for affected models
- [x] ActiveAdmin: verify "Import" action for affected models

### CODE REVIEW

**Comment by workgena** [comment]
Test:
```
CustomersAuth
  #create
    when tag_action is NULL
      allows tag_action_value to be empty
      allows tag_action_value to be present
    when tag_action ID="1"(clear)
      allows tag_action_value to be empty
      allows tag_action_value to be present
    when tag_action is not NULL and not "1"(Clear)
      does not allow tag_action_value to be empty
      does not allow tag_action_value to contain NULL
      does not allow tag_action_value to contain duplicate values
```

File: spec/features/numberlists/edit_numberlist_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require 'spec_helper'
+
+describe 'Edit Nunberlist', type: :feature do
```
Comment (Fivell) [inline spec/features/numberlists/edit_numberlist_spec.rb]: Numberlist :)

**Comment by workgena** [comment]
I fixed test on import for CustomersAuth, Destination and Dialpeer.

But lastly faced with another problem in ActiveAdmin. `Export to CSV` and `Import` have been broken some time ago. Mostly by this migration [20171020164700_number_length.rb](https://github.com/yeti-switch/yeti-web/blob/3b45f2895dac78bcd8e548527b3af3fa80e35f0c/db/migrate/20171020164700_number_length.rb)

I propose to fix it in another Pull Request, because it needs new migration which does not related to current Issue. And fixes in active_admin pages.

Also I suppose some other import-models are broken too. So it needs more time to research and fix. Due to missing/pending tests.

@dmitry-sinina whant do you think?

**Comment by dmitry-sinina** [comment]
@workgena, do it on next request

**Comment by workgena** [comment]
Today changes:

* fix structure.sql
* display tags as `status_tag`

screenshot-2018-2-11 customers auths

**Comment by workgena** [comment]
I'm still looking how to refactor decorators. But I think it can be done later.

**Pull request merged**